README.md

```diff
@@ -2,87 +2,122 @@
 
+This integration only works with the 3.0 version of Private AI's Docker image.
 
 ## Prerequisites
 
-This integration requires an endpoint to make requests against. For instructions, please see the [installation guide](https://docs.private-ai.com/installation/). 
+This integration requires an endpoint to make requests against. For instructions, please see the [Private AI installation guide](https://docs.private-ai.com/installation/).
 
-## Installation 
+## Pre-commit Hook Installation
 
 1. Switch to the repo you want to use the hook in.
-2. Run 'pip install pre-commit'
-3. Create a file called '.pre-commit-config.yaml' and add the following code to it:
+1. Run 'pip install pre-commit'
+1. Create a file called '.pre-commit-config.yaml' and add the following code to it:
+
+    ``` yaml
+    repos:
+      - repo: https://github.com/privateai/pai-pre-commit-hook.git
+        rev: v1.2.1-beta
+        hooks:
+          - id: pii-check
+            args:
+              [
+                "--url",
+                "<URL>",
+                "--env-file-path",
+                "<ENV_FILE_PATH>",
+              ]
+            verbose: true
+
+    ```
+
+1. Replace `<URL>` with the url of where your container is hosted.\
+     eg. [http://localhost:8080/v3/process/text](http://localhost:8080/v3/process_text) for a container running locally.
+1. Create a .env file and add your API_KEY like so:\
+    API_KEY=`<put your API KEY here>`
+1. Replace `<ENV_FILE_PATH>` with the path to your .env file.
+1. Run 'pre-commit install' from inside the git repo where you want to use this hook.
 
+After the above steps, your project structure should look somewhat like:
+
+``` shell
+.
+├── .env
+├── .pre-commit-config.yaml
+└── your-project-dir
+    ├── your-project-file-1
+    └── your-project-file-2
 ```
+
+That's it! You're all set for safe commits!
+
+## Additional Configuration (Optional)
+
+1. If you want to treat only certain entities as PII while ignoring others, you can add the optional argument ```--enabled-entities``` followed by the space-separated entity types you wish to mark as PII. If not specified, the default set of entity types is used. The system currently support these [entity types](https://docs.private-ai.com/entities/).
+2. You can extend the list of detected entities using regular expressions. This can be achieved by setting the optional argument ```--blocked-list``` to a list of regular expressions. The format is a space-separated list of pairs `<ENTITY TYPE> <PATTERN>`. The first element of the pairs (i.e. `<ENTITY TYPE>`) is any string to identify the type of entities while the second element of the pairs (i.e. `<PATTERN>`) is a [Python regular expression](https://docs.python.org/3/library/re.html).
+
+### Sample '.pre-commit-config.yaml'
+
+Here is an example of what your pre-commit-config.yaml may look like:
+
+``` yaml
 repos:
   - repo: https://github.com/privateai/pai-pre-commit-hook.git
-    rev: v1.2.1-beta
+    rev: v1.2-beta
```

Comment:
```suggestion
    rev: v1.2.1-beta
```

Updating the example to the latest version

Reply:
The latest version name is actually decided by us when we make a release, are you saying you prefer to name the release I'll do once this PR is in as '1.2.1-beta'?